lib/msf/base/sessions/winrm_command_shell.rb

```diff
@@ -0,0 +1,149 @@
+# -*- coding: binary -*-
+
+require 'msf/util/signallable_event'
+require 'winrm'
+
+module Msf::Sessions
+  #
+  # This class provides a session for WinRM client connections, where Metasploit
+  # has authenticated to a remote WinRM instance.
+  #
+  class WinrmCommandShell < Msf::Sessions::CommandShell
```

Comment:
So if we're actually running Powershell, I think we should be inheriting from `Msf::Sessions::Powershell`. While Powershell also inherits from `CommandShell`, the important bit is that it will correctly set the session type to `"powershell"` which is important for our post libraries.

Reply:
Done.